kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/IcebergWriterCompatV1Suite.scala

```diff
@@ -399,7 +396,90 @@ class IcebergWriterCompatV1Suite extends DeltaTableWriteSuiteBase with ColumnMap
     expectedErrorMessage =
       "Table features [deletionVectors] are incompatible with icebergCompatV2")
 
-  test("All expected compatible features can be enabled with icebergWriterCompatV1") {
+  /* ----- Non-legacy incompatible features not allowed even when inactive  ----- */
```

Comment:
Previously there was no way to create a table in Kernel with an inactive table feature (since we only supported enabling table features automatically via metadata). https://github.com/delta-io/delta/pull/4340 added support for manually enabling table features so I've added new tests in this suite using the config to enable inactive features